File: 1876-substrings-of-size-three-with-distinct-characters/1876.py
class Solution:
    def isUniq(self, s):
        return len(s) == 3 and s[0] != s[1] != s[2] and s[0] != s[2]

    def countGoodSubstrings(self, s: str) -> int:
        head = 0
        tail = head + 3
        num = 0

        while tail <= len(s):
            if self.isUniq(s[head:tail]):
                num += 1
            head += 1
            tail += 1

        return num

File: 1876-substrings-of-size-three-with-distinct-characters/1876.cpp
class Solution {
public:
    int countGoodSubstrings(string s) {
        int count = 0;
        if (s.size() < 3)
            return 0;
        for (size_t i = 0; i < s.size() - 2; ++i) {
            cout << s[i] << s[i+1] << s[i+2] << endl;
            if (s[i] != s[i + 1] && s[i + 1] != s[i + 2] && s[i] != s[i + 2])
                ++count;
        }
        return count;
    }
};
